lib/base-compiler.js

```diff
@@ -172,12 +179,27 @@ Compile.prototype.compile = function (source, options, filters) {
                         result.asm = "<Compilation failed>";
                         return result;
                     }
+                    result.hasOptOutput = false;
+                    if(self.couldHaveOptOutput(options, self.compiler.version)) {
+                        const optPath = path.join(dirPath, outputFilebase + ".opt.yaml");
```

Comment:
Nice: I need to start using more modern Javascriptisms like `const` and the new labmdas. However...that does mean breaking compatibility with older nodes. I think it's time we did that though!